app/Http/Controllers/AnniversaryController.php

```diff
@@ -0,0 +1,57 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use App\Models\Anniversary;
+use Illuminate\Http\Request;
+use Illuminate\Support\Facades\Auth;
+
+class AnniversaryController extends Controller
+{
+    public function index()
+    {
+        $anniversaries = Auth::user()->anniversaries;
+        return view('pages/anniversary', compact('anniversaries'));
+    }
+
+    public function store(Anniversary $anniversay, Request $request)
+    {
+        $request->validate([
+            'anniversary_date' => 'required|date',
+            'anniversary_type' => 'required|integer|min:0|max:2',
+            'importance' => 'required|integer|min:0|max:3',
+        ]);
```

Comment:
since the validation on the store & update method are the same. you can use the create a laravel request Validation and re-use it in these two places.

check out here:
https://laravel.com/docs/10.x/validation#creating-form-requests